spec/factories/report.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
+  factory :v2_report, class: 'V2::Report' do
+    to_create do |instance, context|
+      policy = FactoryBot.create(
+        :v2_policy,
+        :for_tailoring,
+        title: context.title,
+        system_count: context.assigned_system_count,
+        os_major_version: context.os_major_version,
+        compliance_threshold: context.compliance_threshold,
+        business_objective: context.business_objective,
+        account: context.account,
+        supports_minors: context.supports_minors
+      )
+
+      unsupported_systems = policy.systems.first(context.unsupported_system_count)
+      compliant_systems = (policy.systems - unsupported_systems).first(context.compliant_system_count)
+      other_systems = policy.systems - unsupported_systems - compliant_systems
+
+      unsupported_systems.each do |system|
+        FactoryBot.create(
+          :v2_test_result,
+          system: system,
+          account: context.account,
+          supported: false,
+          score: nil,
+          policy_id: policy.id
+        )
+      end
+
+      compliant_systems.each do |system|
+        FactoryBot.create(
+          :v2_test_result,
+          system: system,
+          account: context.account,
+          supported: true,
+          score: SecureRandom.rand(100 - context.compliance_threshold) + context.compliance_threshold,
```

Comment:
You can create a `score_below` and `score_above` transient pair that could be handled in the TR factory.